mixer/pkg/protobuf/yaml/decoder.go

```diff
@@ -50,48 +56,50 @@ func NewDecoder(resolver Resolver, msgName string, fieldMask map[string]bool) *D
 	}
 
 	return &Decoder{
-		resolver: resolver,
-		fields:   fields,
+		resolver:   resolver,
+		fields:     fields,
+		attrPrefix: attrPrefix,
 	}
 }
 
 // Decode function parses wire-encoded bytes to attribute values. The keys are field names
 // in the message specified by the field mask.
-func (d *Decoder) Decode(b []byte) (map[string]interface{}, error) {
+func (d *Decoder) Decode(b []byte, out *attribute.MutableBag) error {
```

Comment:
reviewers: changing the signature to avoid instantiating maps or copying byte arrays